adapters/socketio_mbed.c

```diff
@@ -189,7 +189,10 @@ void socketio_destroy(CONCRETE_IO_HANDLE socket_io)
     {
         SOCKET_IO_INSTANCE* socket_io_instance = (SOCKET_IO_INSTANCE*)socket_io;
 
-        tcpsocketconnection_destroy(socket_io_instance->tcp_socket_connection);
+        if (socket_io_instance->tcp_socket_connection != NULL)
+        {
+            tcpsocketconnection_destroy(socket_io_instance->tcp_socket_connection);
+        }
```

Comment:
Do you want `socket_io_instance->tcp_socket_connection = NULL;` after destroy, as in Line 242 - 243?
```
                tcpsocketconnection_destroy(socket_io_instance->tcp_socket_connection);
                socket_io_instance->tcp_socket_connection = NULL;
```

Reply:
not needed because socket_io_instance is being freeded